src/compiler/transformers/module/module.ts

```diff
@@ -671,28 +707,53 @@ namespace ts {
 
         function visitExpressionStatement(node: ExpressionStatement): VisitResult<Statement> {
             const original = getOriginalNode(node);
-            if (original.kind === SyntaxKind.EnumDeclaration
-                && hasModifier(original, ModifierFlags.Export)) {
-                return visitExpressionStatementForEnumDeclaration(node, <EnumDeclaration>original);
+            const origKind = original.kind;
+
+            if (origKind === SyntaxKind.EnumDeclaration || origKind === SyntaxKind.ModuleDeclaration) {
+                return visitExpressionStatementForEnumOrNamespaceDeclaration(node, <EnumDeclaration | ModuleDeclaration>original);
+            }
+            else if (origKind === SyntaxKind.ClassDeclaration) {
+                // The decorated assignment for a class name will potentially need to be transformed.
+                const classDecl = original as ClassDeclaration;
+                if (classDecl.name) {
+                    const statements = [node];
+                    // Avoid emitting a default because that will typically be taken care of for us.

```

Comment:
I don't understand this comment -- the subsequent assert makes me think that it only emits a default when there is a decorator. Otherwise an exported non-default would add statement(s?) for both calls to `addExportMemberAssignments`. Can you make the explanation more precise?